seaborn/rcmod.py

```diff
@@ -460,6 +433,35 @@ def set_context(context=None, font_scale=1, rc=None):
     mpl.rcParams.update(context_object)
 
 
+class _StyleOrContext(dict):
+    def __enter__(self):
+        rc = mpl.rcParams
+        self._orig = {k: rc[k] for k in self._keys}
+        type(self)._set(self)
```

Comment:
Because `self._set` would treat (on Py3) `_set` as a method and fill in the first argument with `self`, see for example
```
In [1]: def f(*args): print(args)

In [2]: class C:
    method = f
   ...: 

In [3]: C().method()
(<__main__.C object at 0x7f9538812d30>,)
```
Either I need to call via `type` (to avoid binding self to the first argument), or I can do `_set = staticmethod(set_keys)` further down -- either way is fine with me.